### PR TITLE
Update Drosophila epithelia cell boundary segmentation of 2D projections

### PR DESCRIPTION
**Pull request by bioimageiobot**
This is an automatic PR created by the @bioimageiobot regarding changes to the resource item 10.5281/zenodo.7380171.
The following version(s) will be added:
  - [7405349 (version 3)](https://www.doi.org/10.5281/zenodo.7405349)

Please review the changes and make sure the new item or version(s) pass the following check list:
- [ ] Passed the BioImage.IO CI tests: static (and dynamic) validations
- [ ] The meta information for the RDF item is complete
  - [ ] The tags are complete and describe the model
  - [ ] Naming is intuitive and descriptive, example:  Multi-Organ Nucleus Segmentation (StarDist 2D)
  - [ ] Authors are provided
  - [ ] Documentation is complete
    * For models, include an overview, describe how the model is trained, what is the training data, how to use the model, how to validate the results and list the references. TODO: Model documentation template.
- [ ] Approved by at least one of the BioImage.IO admin team member.

Maintainers: none specified

Note: If you updated or re-uploaded another version for the current item on Zenodo, this PR won't be changed automatically. To proceed, you can do the following:
 1. Block this version, but keep looking for future versions: Edit the current resource.yaml and keep the top-level `status` field as `accepted`, but change the `status` under the current version to `blocked`.
 2. Accept this version and keep looking for future versions: Merge this PR for now.
 3. Keep proposed version(s) (and this resource in general if it is new) as pending: Close this PR without merging.

 Then wait for the CI on the main branch to complete. It should detect the new version(s) and create another PR for the new version(s).

Previous PRs of this resource: https://github.com/bioimage-io/collection-bioimage-io/pull/510, https://github.com/bioimage-io/collection-bioimage-io/pull/514, https://github.com/bioimage-io/collection-bioimage-io/pull/516

### CODE REVIEW

**Comment by odinsbane** [comment]
I am not sure what the error is asking for. It seems like something cannot be downloaded? We are going to publish this work relatively soon. Can we put some actual sample images into the zenodo repository or will that affect the chance of beion on bioimage.io? Thank you.

**Comment by FynnBe** [comment]
> Can we put some actual sample images into the zenodo repository or will that affect the chance of beion on bioimage.io?

Yes of course. In this case a missing `cover.png` file would even prevent you from contributing...

**Comment by FynnBe** [comment]
@odinsbane could you share a conda environment that your model works in?
Then I'll check if the issue can be fixed on our side. 

Otherwise you can just add it to the zenodo record (yet another version) and link it in the `weights` section to each weights format,lik so:
```yaml
weights:
  tensorflow_saved_model_bundle:
    ...
    dependencies: conda:environment.yaml
````

**Comment by odinsbane** [comment]
The model has worked on all of the tensorflow versions I've tried. 2.4, 2.5, 2.7, 2.8 and 2.9.  

I can make sure they work with a tensorflow version that you suggest. I'm not sure which version is failing though, the h5 or the zip file? 

I have one conda environment that I use. It's a bit old, otherwise I just use pip + venv. 

```
name: tf24
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _tflow_select=2.1.0=gpu
  - absl-py=0.15.0=pyhd3eb1b0_0
  - aiohttp=3.8.1=py37h7f8727e_0
  - aiosignal=1.2.0=pyhd3eb1b0_0
  - astor=0.8.1=py37h06a4308_0
  - astunparse=1.6.3=py_0
  - async-timeout=4.0.1=pyhd3eb1b0_0
  - asynctest=0.13.0=py_0
  - attrs=21.4.0=pyhd3eb1b0_0
  - blas=1.0=mkl
  - blinker=1.4=py37h06a4308_0
  - brotlipy=0.7.0=py37h27cfd23_1003
  - c-ares=1.18.1=h7f8727e_0
  - ca-certificates=2022.2.1=h06a4308_0
  - cachetools=4.2.2=pyhd3eb1b0_0
  - certifi=2021.10.8=py37h06a4308_2
  - cffi=1.15.0=py37hd667e15_1
  - charset-normalizer=2.0.4=pyhd3eb1b0_0
  - click=8.0.4=py37h06a4308_0
  - cryptography=3.4.8=py37hd23ed53_0
  - cudatoolkit=10.1.243=h6bb024c_0
  - cudnn=7.6.5=cuda10.1_0
  - cupti=10.1.168=0
  - dataclasses=0.8=pyh6d0b6a4_7
  - frozenlist=1.2.0=py37h7f8727e_0
  - gast=0.4.0=pyhd3eb1b0_0
  - google-auth=1.33.0=pyhd3eb1b0_0
  - google-auth-oauthlib=0.4.4=pyhd3eb1b0_0
  - google-pasta=0.2.0=pyhd3eb1b0_0
  - grpcio=1.42.0=py37hce63b2e_0
  - h5py=2.10.0=py37hd6299e0_1
  - hdf5=1.10.6=hb1b8bf9_0
  - idna=3.3=pyhd3eb1b0_0
  - importlib-metadata=4.8.2=py37h06a4308_0
  - intel-openmp=2021.4.0=h06a4308_3561
  - keras-preprocessing=1.1.2=pyhd3eb1b0_0
  - ld_impl_linux-64=2.35.1=h7274673_9
  - libffi=3.3=he6710b0_2
  - libgcc-ng=9.1.0=hdf63c60_0
  - libgfortran-ng=7.3.0=hdf63c60_0
  - libprotobuf=3.19.1=h4ff587b_0
  - libstdcxx-ng=9.1.0=hdf63c60_0
  - markdown=3.3.4=py37h06a4308_0
  - mkl=2021.4.0=h06a4308_640
  - mkl-service=2.4.0=py37h7f8727e_0
  - mkl_fft=1.3.1=py37hd3c417c_0
  - mkl_random=1.2.2=py37h51133e4_0
  - multidict=5.2.0=py37h7f8727e_2
  - ncurses=6.2=he6710b0_1
  - numpy=1.21.2=py37h20f2e39_0
  - numpy-base=1.21.2=py37h79a1101_0
  - oauthlib=3.2.0=pyhd3eb1b0_0
  - openssl=1.1.1m=h7f8727e_0
  - opt_einsum=3.3.0=pyhd3eb1b0_1
  - pip=21.1.3=py37h06a4308_0
  - protobuf=3.19.1=py37h295c915_0
  - pyasn1=0.4.8=pyhd3eb1b0_0
  - pyasn1-modules=0.2.8=py_0
  - pycparser=2.21=pyhd3eb1b0_0
  - pyjwt=2.1.0=py37h06a4308_0
  - pyopenssl=21.0.0=pyhd3eb1b0_1
  - pysocks=1.7.1=py37_1
  - python=3.7.11=h12debd9_0
  - python-flatbuffers=2.0=pyhd3eb1b0_0
  - readline=8.1=h27cfd23_0
  - requests=2.27.1=pyhd3eb1b0_0
  - requests-oauthlib=1.3.0=py_0
  - rsa=4.7.2=pyhd3eb1b0_1
  - scipy=1.6.2=py37had2a1c9_1
  - setuptools=52.0.0=py37h06a4308_0
  - six=1.16.0=pyhd3eb1b0_1
  - sqlite=3.36.0=hc218d9a_0
  - tensorboard=2.4.0=pyhc547734_0
  - tensorboard-plugin-wit=1.6.0=py_0
  - tensorflow=2.4.1=gpu_py37ha2e99fa_0
  - tensorflow-base=2.4.1=gpu_py37h29c2da4_0
  - tensorflow-estimator=2.6.0=pyh7b7c402_0
  - tensorflow-gpu=2.4.1=h30adc30_0
  - termcolor=1.1.0=py37h06a4308_1
  - tk=8.6.10=hbc83047_0
  - typing-extensions=3.10.0.2=hd3eb1b0_0
  - typing_extensions=3.10.0.2=pyh06a4308_0
  - urllib3=1.26.8=pyhd3eb1b0_0
  - werkzeug=2.0.3=pyhd3eb1b0_0
  - wheel=0.36.2=pyhd3eb1b0_0
  - wrapt=1.13.3=py37h7f8727e_2
  - xz=5.2.5=h7b6447c_0
  - yarl=1.6.3=py37h27cfd23_0
  - zipp=3.7.0=pyhd3eb1b0_0
  - zlib=1.2.11=h7b6447c_3
  - pip:
    - cycler==0.11.0
    - fonttools==4.29.1
    - imageio==2.16.1
    - kiwisolver==1.3.2
    - matplotlib==3.5.1
    - networkx==2.6.3
    - packaging==21.3
    - pillow==9.0.1
    - pyparsing==3.0.7
    - python-dateutil==2.8.2
    - pywavelets==1.2.0
    - scikit-image==0.18.3
    - tifffile==2021.11.2
    - urwid==2.1.2
prefix: ###
```

**Comment by odinsbane** [comment]
I see, the model bundle doesn't load with tensorflow 2.4. I think I used a new version of tensorflow to convert the .h5 file to a model bundle format. I think the easiest would be to update the version in the yaml file? Otherwise I can save .h5 file as a model bundle with an older tensorflow?

**Comment by FynnBe** [comment]
Didn't want to try out several tensorflow versions for the model bundle, so I set it to tf 2.10 and hope that'll work...

**Comment by odinsbane** [comment]
So now it appears to be trying to load the model bundle as an h5 file. I dont' mind making another change to the zenodo repository to make this work smoother. If there is a tensorflow config is recommended, I can try to make it work on my end first. The h5 file was trained with 2.4/2.5 so I should be able to open that with any version in between. I will include the pip freeze of the environment I used to create the bioimage repo.

```
(bioimage-io-env) [username@localhost bioimage-prep]$ pip freeze 
absl-py==1.3.0
astunparse==1.6.3
bioimageio.core==0.5.7
bioimageio.spec==0.4.8.post1
cachetools==5.2.0
certifi==2022.9.24
charset-normalizer==2.1.1
click==8.1.3
configparser==5.3.0
contourpy==1.0.6
csbdeep==0.7.2
cycler==0.11.0
flatbuffers==1.12
fonttools==4.38.0
funcsigs==1.0.2
gast==0.4.0
google-auth==2.14.1
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
gputools==0.2.13
grpcio==1.50.0
h5py==3.7.0
idna==3.4
imageio==2.22.4
importlib-metadata==5.1.0
keras==2.9.0
Keras-Preprocessing==1.1.2
kiwisolver==1.4.4
libclang==14.0.6
llvmlite==0.39.1
Mako==1.2.4
Markdown==3.4.1
MarkupSafe==2.1.1
marshmallow==3.19.0
marshmallow-jsonschema==0.13.0
marshmallow-union==0.1.15.post1
matplotlib==3.6.2
networkx==2.8.8
numba==0.56.4
numpy==1.23.4
oauthlib==3.2.2
opt-einsum==3.3.0
packaging==21.3
pandas==1.5.1
Pillow==9.3.0
platformdirs==2.5.4
protobuf==3.19.6
pyasn1==0.4.8
pyasn1-modules==0.2.8
pyopencl==2022.3
pyparsing==3.0.9
python-dateutil==2.8.2
pytools==2022.1.13
pytz==2022.6
PyWavelets==1.4.1
reikna==0.7.6
requests==2.28.1
requests-oauthlib==1.3.1
rsa==4.9
ruamel.yaml==0.17.21
ruamel.yaml.clib==0.2.7
scikit-image==0.19.3
scipy==1.9.3
six==1.16.0
stardist==0.8.3
style==1.1.0
tensorboard==2.9.1
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow==2.9.0
tensorflow-estimator==2.9.0
tensorflow-io-gcs-filesystem==0.28.0
termcolor==2.1.1
tifffile==2022.10.10
tqdm==4.64.1
typer==0.7.0
typing_extensions==4.4.0
urllib3==1.26.12
Werkzeug==2.2.2
wrapt==1.14.1
xarray==2022.11.0
zipp==3.11.0
```

**Comment by FynnBe** [comment]
there is still an issue on our side. We end up installing tf 2.1 and not 2.10 as now specified for the tf model bundle weights...

**Comment by odinsbane** [comment]
Do you know if the validation is using the GPU? When I was setting this up, I tried it without the gpu and I had the error that is being reported. When I used the GPU it validated.

**Comment by FynnBe** [comment]
I had wrongly specified the tensorflow version as a float, which lead to 2.10 being truncated to 2.1.

**Comment by FynnBe** [comment]
ok, we are close to resolving the dynamic test case failiure!
current error (with tf 2.10):
```
'Mismatched elements: 3868 / 12288 (31.5%)\n'
            'Max absolute difference: 0.00392532\n'
            'Max relative difference: 0.14776838\n'
```
I'll try tensorflow_version 2.9 and 2.8 before decreasing the comparison accuracy.

**Comment by FynnBe** [comment]
> Do you know if the validation is using the GPU? When I was setting this up, I tried it without the gpu and I had the error that is being reported. When I used the GPU it validated.

here we only have cpu available. This might explain the differences. If tf 2.9 fails I'll decrease test accuracy...

**Comment by FynnBe** [comment]
Yay! 🎉

**Comment by FynnBe** [comment]
@odinsbane  I added you as a maintainer of the model. If that's ok, I'll merge this PR and your model will find it's way onto bioimage.io !

**Comment by odinsbane** [comment]
Excellent! Thank you for getting this sorted!
